sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/implementation/Batcher.java

```diff
@@ -137,18 +143,43 @@ private LogsIngestionRequest nextInternal() throws IOException {
         return result;
     }
 
+    /*private LogsIngestionRequest nextInternal() throws IOException {
+        LogsIngestionRequest result = null;
+        while (iterator.hasNext() && result == null) {
+            Object currentLog = iterator.next();
+            byte[] bytes = serializer.serializeToBytes(currentLog);
+            currentBatchSize += bytes.length;
+            if (currentBatchSize > MAX_REQUEST_PAYLOAD_SIZE) {
+                result = createRequest(false);
+                currentBatchSize = bytes.length;
+            }
+
+            serializedLogs.add(new String(bytes, StandardCharsets.UTF_8));
+            originalLogsRequest.add(currentLog);
+        }
+
+        if (result == null && currentBatchSize > 0) {
+            currentBatchSize = 0;
+            return createRequest(true);
+        }
+
+        return result;
+    }*/
+
     private LogsIngestionRequest createRequest(boolean last) throws IOException {
         ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-        JsonGenerator generator = JsonFactory.builder().build().createGenerator(byteArrayOutputStream);
+        JsonWriter writer = JsonProviders.createWriter(byteArrayOutputStream);
         try {
-            generator.writeStartArray();
-            generator.writeRaw(serializedLogs.stream().collect(Collectors.joining(",")));
-            generator.writeEndArray();
-            generator.close();
-
+            writer.writeStartArray();
+            for (String log : serializedLogs) {
+                writer.writeRawValue(log);
+            }
```

Comment:
If this ends up causing problems this could be changed to use a single `writeRawValue` with string joining like it was before (but this is a better choice as it reduces string concatenations going on).